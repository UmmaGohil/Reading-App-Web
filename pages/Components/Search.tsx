import React from "react";
import Page from "../Pages/Page";
import CSS from "csstype";

interface Props {
	onChange(e: React.ChangeEvent<HTMLInputElement>): void;
}

const input: CSS.Properties = {
	width: "100%",
	height: "5rem",
	color: "#2938fa",
	backgroundColor: "#f1a5d0",
	border: "none",
	fontSize: "2rem",
	margin: "0",
	textAlign: "center",
};

const Search = ({ onChange }: Props) => (
	<Page variant="regular">
		<input
			type="search"
			placeholder="SEARCH"
			className="search"
			onChange={onChange}
			style={input}
		/>
	</Page>
);

export default Search;
