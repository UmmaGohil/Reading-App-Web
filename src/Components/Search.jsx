import React from 'react';
import Page from '../Pages/Page'

const Search = () => 
  <Page variant="regular">
    <form>
      <input type="text" />
      <button>Go</button>
    </form>
  </Page>

export default Search

