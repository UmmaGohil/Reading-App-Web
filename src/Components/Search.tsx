import React from 'react'
import Page from '../Pages/Page'
import CSS from 'csstype'

interface Props {
  onChange(e: any): void 
}

const input: CSS.Properties = {
  width: '100%',
  height: '5rem',
  color: '#2938fa',
  backgroundColor: '#f1a5d0',
  border: 'none', 
  fontSize: '2rem'
};

const Search = ({ onChange }: Props) => 
  <Page variant="regular">
    <form>
      <input 
        type='search'
        placeholder='Search...'
        className='search'
        onChange={onChange}
        style={input}
      />
    </form>
  </Page>

export default Search

