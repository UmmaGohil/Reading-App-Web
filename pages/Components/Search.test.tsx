/*import * as React from "react";
import { render, screen } from "@testing-library/react";
import Search from "./Search";
import { isVoidExpression } from "typescript";

describe("<Search />", () => {
	beforeEach(() => {
		render(<Search onChange={() => {}} />);
	});
	it("find inputs", () => {
		expect(screen.getAllByRole("searchbox")).toHaveLength(1);
	});
});*/

import { render } from "@testing-library/react";
import { expect, it } from "vitest";
import Search from "./Search";

it("renders correctly", () => {
	const result = render(<Search onChange={() => {}} />);
	expect(result).toMatchSnapshot();
});
