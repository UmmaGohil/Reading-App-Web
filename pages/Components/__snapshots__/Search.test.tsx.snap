// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        class="search"
        placeholder="SEARCH"
        style="width: 100%; height: 5rem; color: rgb(41, 56, 250); background-color: rgb(241, 165, 208); font-size: 2rem; margin: 0px; text-align: center;"
        type="search"
      />
    </div>
  </body>,
  "container": <div>
    <input
      class="search"
      placeholder="SEARCH"
      style="width: 100%; height: 5rem; color: rgb(41, 56, 250); background-color: rgb(241, 165, 208); font-size: 2rem; margin: 0px; text-align: center;"
      type="search"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
